pir/cpp/client_test.cpp

```diff
@@ -68,4 +78,69 @@ TEST_F(PIRClientTest, TestProcessResponse) {
   ASSERT_EQ(result, value);
 }
 
+TEST_F(PIRClientTest, TestCreateRequest_InvalidIndex) {
+  auto payload_or = client_->CreateRequest(DB_SIZE + 1);
+  ASSERT_EQ(payload_or.status().code(),
+            private_join_and_compute::StatusCode::kInvalidArgument);
+}
+
+class CreateRequestTest
+    : public PIRClientTest,
+      public testing::WithParamInterface<tuple<size_t, size_t, uint64_t>> {};
+
+TEST_P(CreateRequestTest, TestCreateRequest_MoreThanOneCT) {
+  const auto dbsize = get<0>(GetParam());
+  int desired_index = get<1>(GetParam());
+  SetUpDB(dbsize);
+
+  const auto poly_modulus_degree =
+      pir_params_->GetEncryptionParams().poly_modulus_degree();
+  const auto plain_mod =
+      pir_params_->GetEncryptionParams().plain_modulus().value();
+
+  auto payload_or = client_->CreateRequest(desired_index);
+  ASSERT_TRUE(payload_or.ok())
+      << "Status is: " << payload_or.status().ToString();
+  auto payload = payload_or.ValueOrDie();
+  ASSERT_EQ(payload.Get().size(), dbsize / poly_modulus_degree + 1);
+
+  for (const auto& ct : payload.Get()) {
+    Plaintext pt;
+    Decryptor()->decrypt(ct, pt);
+    // std::cout << "CreateRequest PT: " << pt.to_string() << std::endl;
```

Comment:
Please delete or uncomment.

Reply:
I was actually thinking of trying to hide these behind a flag or something for debugging, but didn't get around to it. WDYT?